### ISSUE TITLE
android 11 error

### ISSUE THREAD

**Issue by 424344** [asksven/BetterBatteryStats#876]
Hello,
I'm using beta  2.6-378 on android 11 stock (not rooted).
the device is a samsung a42 5g.

I have installed the apk and given the permissions using adb, but it doesn't work.
See the screenshot for the error (sorry the screenshot is in italian, I have selected to force english language in bbs options but it is still displayed in italian).

On another device running android 10 it works correctly.

![Screenshot_20210330-113638](https://user-images.githubusercontent.com/26486758/112973368-871aef80-9151-11eb-84f5-d5272d989c19.jpg)

**Comment by tortino** [comment]
Here are debug logs.

[dmesg-2021-04-08_144811866.txt](https://github.com/asksven/BetterBatteryStats/files/6280443/dmesg-2021-04-08_144811866.txt)
[logcat-2021-04-08_144811512.txt](https://github.com/asksven/BetterBatteryStats/files/6280444/logcat-2021-04-08_144811512.txt)
[BetterBatteryStats-2021-04-08_144811472.txt](https://github.com/asksven/BetterBatteryStats/files/6280446/BetterBatteryStats-2021-04-08_144811472.txt)

**Comment by tortino** [comment]
hello,
I have searched for a solution for the "transaction failed on small parcel" error.

I found something here:
https://developer.android.com/guide/app-compatibility/restrictions-non-sdk-interfaces
https://stackoverflow.com/questions/55970137/bypass-androids-hidden-api-restrictions

bbs it seems to work if I use this command:
adb shell settings put global hidden_api_policy 1

**Comment by joanroig** [comment]
@tortino I confirm this solution works, thanks

**Comment by asksven** [comment]
I can confirm, see the comments for the last build on xda

On Jun 23, 2021, 20:14, at 20:14, Joan ***@***.***> wrote:
***@***.*** I confirm this solution works, thanks

**Comment by tbertels** [comment]
Is there a way to do that automatically by BBS when rooted? Same thing for the permissions.

**Comment by Milor123** [comment]
> Is there a way to do that automatically by BBS when rooted? Same thing for the permissions.

Instead you can use:

1) Open termux or your terminal and type:
```
su
settings put global hidden_api_policy 1
```

2) Force to close BBS app 
3) Start BBS again
then problem solved!!!

I am not sure if it works without root and also not know about if after reboot is restored to default value but you could use smart pack kernel app for automatize launch script when reboot you phone.

It work for me in android 11 Thank you very much guys

**Comment by skpaul2000** [comment]
Thanks, this solution worked! I was facing the issue in my Redmi Note 10. A reboot is necessary after using the adb command, for bbs to start working.

**Comment by asksven** [comment]
Instructions on xda and in-app updated

**Comment by mirh** [comment]
I updated from 420 to 426, and I was no longer getting the parcel error (`Success (fallback)` having replaced it). 

But kernel wakelocks were still nowhere to be found, and after casually checking with `settings get global hidden_api_policy_p_apps`.. guess what? The value was null, unset. 
(now that I set it everything works perfectly). 

So I think there must be a bug, or at least a shortcoming, in your hasPermissionToCallHiddenApis check.
